music21/expressions.py

```diff
@@ -517,21 +519,205 @@ def fillListOfRealizedNotes(
+                        pitchPastMeasure=pitchPastMeasure,
+                        alteredPitches=alteredPitches,
+                        cautionaryPitchClass=cautionaryPitchClass,
+                        cautionaryAll=cautionaryAll,
+                        overrideStatus=overrideStatus,
+                        cautionaryNotImmediateRepeat=cautionaryNotImmediateRepeat,
+                        lastNoteWasTied=lastNoteWasTied)
+
+            # then lower
+            p = self.lowerPitch
+            if p is not None:
+                if self.lowerAccid is not None:
+                    # force a visible lower accidental
+                    if p.accidental is None:
+                        p.accidental = pitch.Accidental(0)
+                    p.accidental.displayStatus = True
+                else:
+                    p.updateAccidentalDisplay(
+                        pitchPast=pitchPast,
+                        pitchPastMeasure=pitchPastMeasure,
+                        alteredPitches=alteredPitches,
+                        cautionaryPitchClass=cautionaryPitchClass,
+                        cautionaryAll=cautionaryAll,
+                        overrideStatus=overrideStatus,
+                        cautionaryNotImmediateRepeat=cautionaryNotImmediateRepeat,
+                        lastNoteWasTied=lastNoteWasTied)
+            return
+
+        if isinstance(self, (GeneralMordent, Trill)):
+            # just one otherPitch
+            p = self.otherPitches[0]
+            if p is not None:
+                if self.accid is not None:
+                    # force a visible accidental
+                    if p.accidental is None:
+                        p.accidental = pitch.Accidental(0)
+                    p.accidental.displayStatus = True
+                else:
+                    p.updateAccidentalDisplay(
+                        pitchPast=pitchPast,
+                        pitchPastMeasure=pitchPastMeasure,
+                        alteredPitches=alteredPitches,
+                        cautionaryPitchClass=cautionaryPitchClass,
+                        cautionaryAll=cautionaryAll,
+                        overrideStatus=overrideStatus,
+                        cautionaryNotImmediateRepeat=cautionaryNotImmediateRepeat,
+                        lastNoteWasTied=lastNoteWasTied)
+
+    @property
+    def otherPitches(self) -> tuple[pitch.Pitch, ...]:
+        return self._otherPitches
+
 
 # ------------------------------------------------------------------------------
 class GeneralMordent(Ornament):
     '''
     Base class for all Mordent types.
     '''
-    def __init__(self, **keywords):
+    def __init__(self, *, accid: str | None = None, **keywords):
         super().__init__(**keywords)
-        self.direction = ''  # up or down
-        self.size = None  # interval.Interval (General, etc.) class
+        self._direction = ''  # up or down
+        self._accid: str | None = None
```

Comment:
(_accid to _accidentalName or _accidentalStr)  -- no abbreviations.

Why can't this use '' as a None value?  It's much easier for users in IDEs to have a single type as an answer.  Then just `if accidentalName...` latter.

Reply:
Done.